src/ansys/templates/python/common/{{cookiecutter.__project_name_slug}}/pyproject.toml

```diff
@@ -65,6 +65,25 @@ packages = [
 [tool.poetry.dependencies]
 python = ">={{ cookiecutter.__requires_python }},<4.0"
 importlib-metadata = {version = "^4.0", python = "<3.8"}
+
+# Optional documentation dependencies
+Sphinx = {version = "==5.1.1", optional = true}
+numpydoc = {version = "==1.4.0", optional = true}
+ansys-sphinx-theme = {version = "==0.4.2", optional = true}
+sphinx-copybutton = {version = "==0.5.0", optional = true}
+
+# Optional testing dependencies
+pytest = {version = "==7.1.2", optional = true}
+pytest-cov = {version = "==3.0.0", optional = true}
+
+# Optional build requirements
+build = {version = "==0.8.0", optional = true}
+twine = {version = "==4.0.1", optional = true}
+
+[tool.poetry.extras]
+dist = ["build", "twine"]
+tests = ["pytest", "pytest-cov"]
+doc = ["Sphinx", "numpydoc", "ansys-sphinx-theme", "sphinx-copybutton"]
```

Comment:
Unless the end-user actually needs to install these dependencies, I find the new "group-based" syntax of poetry nicer. Mostly, because it avoids the abundance of `{version=, optional=true}`.

These will be visible only during development, and not included in the distributed wheel.

As a side note, since the `poetry.lock` already locks these dev dependencies, do they need to be pinned in the `pyproject.toml`? I'd guess it's nice if dependabot gets to update these, too.

```suggestion

# Optional documentation dependencies
[tool.poetry.group.doc]
optional = true

[tool.poetry.group.doc.dependencies]
Sphinx = "==5.1.1"
numpydoc = "==1.4.0"
ansys-sphinx-theme = "==0.4.2"
sphinx-copybutton = "==0.5.0"

# Optional testing dependencies
[tool.poetry.group.tests]
optional = true
[tool.poetry.group.tests.dependencies]
pytest = "==7.1.2"
pytest-cov = "==3.0.0"

# Optional build requirements
[tool.poetry.group.build]
optional = true
[tool.poetry.group.build.dependencies]
build = "==0.8.0"
twine = "==4.0.1"
```

Reply:
Glad to see group dependencies are in the latest stable version for poetry, now. Thanks for suggesting these changes, @greschd.

Reply:
From @greschd comment, I agree on removing the version and assume it is going to be managed by the `poetry.lock` file.